### ISSUE TITLE
SMTP Timeout

### ISSUE THREAD

**Issue by wutzi15** [uxbox/uxbox#2292]
**To Reproduce**

1. Try inviting a new user by email

**Actual behavior**

The sending of the email files failes with a time out. See Stack trace below.  There is no error message in the frontend. The smtp config I use works for Thunderbird, Apple Mail and Outlook. Setting PENPOT_FLAGS="... disable-smtp" does not print the email to the log, but also times out with the same stack trace.
 
**Environment (please complete the following information):**
- Host (e.g. https://design.penpot.app, local instance): local instance

PENPOT_FLAGS="enable-login enable-smtp"

PENPOT_SMTP_ENABLED=true
PENPOT_SMTP_DEFAULT_FROM=mail@example.com
PENPOT_SMTP_DEFAULT_REPLY_TO=mail@example.com
PENPOT_SMTP_HOST=smtp.example.com
PENPOT_SMTP_PORT=465
ENPOT_SMTP_USERNAME=mail@example.com
PENPOT_SMTP_PASSWORD=FOOBAR
#PENPOT_SMTP_TLS=true
PENPOT_SMTP_SSL=true

*If self-hosted:*
- OS Version (e.g. Ubuntu 16.04):
- Docker / Docker-compose version (e.g. Docker version 18.03.0-ce, build 0520e24): 20.10.18
- Image version (e.g. Alpine): latest

Backend Stack Trace:
<details>
penpot-penpot-backend-1   | jakarta.mail.MessagingException: Exception reading response
penpot-penpot-backend-1   | 	at com.sun.mail.smtp.SMTPTransport.readServerResponse(SMTPTransport.java:2476) ~[penpot.jar:?]
penpot-penpot-backend-1   | 	at com.sun.mail.smtp.SMTPTransport.openServer(SMTPTransport.java:2172) ~[penpot.jar:?]
penpot-penpot-backend-1   | 	at com.sun.mail.smtp.SMTPTransport.protocolConnect(SMTPTransport.java:722) ~[penpot.jar:?]
penpot-penpot-backend-1   | 	at jakarta.mail.Service.connect(Service.java:342) ~[penpot.jar:?]
penpot-penpot-backend-1   | 	at jakarta.mail.Service.connect(Service.java:222) ~[penpot.jar:?]
penpot-penpot-backend-1   | 	at jakarta.mail.Service.connect(Service.java:243) ~[penpot.jar:?]
penpot-penpot-backend-1   | 	at jakarta.mail.Transport.send0(Transport.java:228) ~[penpot.jar:?]
penpot-penpot-backend-1   | 	at jakarta.mail.Transport.send(Transport.java:150) ~[penpot.jar:?]
penpot-penpot-backend-1   | 	at app.util.emails$send_BANG_.invokeStatic(emails.clj:172) ~[penpot.jar:?]
penpot-penpot-backend-1   | 	at app.util.emails$send_BANG_.invoke(emails.clj:169) ~[penpot.jar:?]
penpot-penpot-backend-1   | 	at app.emails$fn__27382$fn__27384.invoke(emails.clj:170) ~[penpot.jar:?]
penpot-penpot-backend-1   | 	at app.worker$wrap_task_handler$fn__17949.invoke(worker.clj:553) ~[penpot.jar:?]
penpot-penpot-backend-1   | 	at app.worker$handle_task.invokeStatic(worker.clj:338) ~[penpot.jar:?]
penpot-penpot-backend-1   | 	at app.worker$handle_task.invoke(worker.clj:334) ~[penpot.jar:?]
penpot-penpot-backend-1   | 	at app.worker$run_task.invokeStatic(worker.clj:383) ~[penpot.jar:?]
penpot-penpot-backend-1   | 	at app.worker$run_task.invoke(worker.clj:375) ~[penpot.jar:?]
penpot-penpot-backend-1   | 	at clojure.core$partial$fn__5910.invoke(core.clj:2647) ~[penpot.jar:?]
penpot-penpot-backend-1   | 	at promesa.util.SupplierWrapper.get(util.cljc:37) ~[penpot.jar:?]
penpot-penpot-backend-1   | 	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768) ~[?:?]
penpot-penpot-backend-1   | 	at java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760) ~[?:?]
penpot-penpot-backend-1   | 	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) ~[?:?]
penpot-penpot-backend-1   | 	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) ~[?:?]
penpot-penpot-backend-1   | 	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) ~[?:?]
penpot-penpot-backend-1   | 	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) ~[?:?]
penpot-penpot-backend-1   | 	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) ~[?:?]
penpot-penpot-backend-1   | Caused by: java.net.SocketTimeoutException: Read timed out
penpot-penpot-backend-1   | 	at sun.nio.ch.NioSocketImpl.timedRead(NioSocketImpl.java:280) ~[?:?]
penpot-penpot-backend-1   | 	at sun.nio.ch.NioSocketImpl.implRead(NioSocketImpl.java:306) ~[?:?]
penpot-penpot-backend-1   | 	at sun.nio.ch.NioSocketImpl.read(NioSocketImpl.java:347) ~[?:?]
penpot-penpot-backend-1   | 	at sun.nio.ch.NioSocketImpl$1.read(NioSocketImpl.java:800) ~[?:?]
penpot-penpot-backend-1   | 	at java.net.Socket$SocketInputStream.read(Socket.java:966) ~[?:?]
penpot-penpot-backend-1   | 	at com.sun.mail.util.TraceInputStream.read(TraceInputStream.java:102) ~[penpot.jar:?]
penpot-penpot-backend-1   | 	at java.io.BufferedInputStream.fill(BufferedInputStream.java:244) ~[?:?]
penpot-penpot-backend-1   | 	at java.io.BufferedInputStream.read(BufferedInputStream.java:263) ~[?:?]
penpot-penpot-backend-1   | 	at com.sun.mail.util.LineInputStream.readLine(LineInputStream.java:100) ~[penpot.jar:?]
penpot-penpot-backend-1   | 	at com.sun.mail.smtp.SMTPTransport.readServerResponse(SMTPTransport.java:2456) ~[penpot.jar:?]
penpot-penpot-backend-1   | 	... 24 more

</details>

**Comment by niwinz** [comment]
I'm not clearly understand the issue. You are trying to configure the smtp or want to print email to console?

**Comment by niwinz** [comment]
In any case, please look on this: https://github.com/penpot/penpot/issues/2249
Many of this issues of default config will be fixed in the next release.

**Comment by wutzi15** [comment]
I'm sorry for the unclear description. It is the end of a long workday for me... 

I'm trying to send an email via SMTP. Printing to STDOUT would just have been a temporary workaround for that.

**Comment by niwinz** [comment]
Yeah, I understand.

With the current version, you need to use the `enable-log-emails` flag to log the emails to console.
With the smtp configuration I don't know how to help, there are so many possible errors and almost all are about setting proper values on the configuration: ensure you using the correct host/port and correct combination of ssl or tls options. If the server lauches timeout on connect (the case you reporting I guess) looks like incorrect configuration

**Comment by wutzi15** [comment]
The wrong configuration is my guess also, however I checked and cross checked it with Apple Mail, Outlook and Thunderbird and the copied and pasted values work in all three of them. Is there any way of debugging that you can recommend?

**Comment by niwinz** [comment]
```bash
openssl s_client -connect  host:port -starttls smtp
```
If this prints something like this:

```
New, TLSv1.2, Cipher is ECDHE-RSA-AES256-GCM-SHA384
Server public key is 2048 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : ECDHE-RSA-AES256-GCM-SHA384
    Session-ID: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
    Session-ID-ctx: 
    Master-Key: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    Start Time: 1663598549
    Timeout   : 7200 (sec)
    Verify return code: 0 (ok)
    Extended master secret: yes
---
250 Ok
```
means you need to use _TLS=true instead of _SSL=true.
I'm pretty sure that your SMTP provider uses TLS=true than SSL, but it also probable that it offers the both in different ports.
You can try the 587 port also (default for tls).

**Comment by cajus** [comment]
I've something that looks like the same issue. I've SSL on port 465 on our mail server. Not TLS. Setting `ENABLE_SSL=true` makes penpot (provided docker compose) run into a timeout. Same timeout for

```
openssl s_client -connect  host:465 -starttls smtp
```

, while

```
openssl s_client -connect  host:port 465
```

works. Isn't there some `mail.smtp.ssl.enable` missing in `app.emails`?

**Comment by niwinz** [comment]
i will check

**Comment by niwinz** [comment]
Yep, this is clearly a bug in our email sending. We have not reached this, because we never used ssl (always tls). It will be fixed in next release (1.16.x). Until now a good workaround is use a local relay; I mean install exim4 and configure it to send emails using your provider, and connect penpot to this localhost exim4. (This approach we use on our production environment, for not connecting penpot directly to the smtp provider)

**Comment by cajus** [comment]
Ok. Thanks @niwinz!
